### ISSUE TITLE
Backup Plan deletion fails randomly

### ISSUE THREAD

**Issue by gui-don** [hashicorp/terraform-provider-aws#10414]
Hi!

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

Terraform v0.11.14
provider.aws v2.19.0
provider.random v2.2.1

### Affected Resource(s)

* aws_backup_selection
* aws_backup_plan

### Terraform Configuration Files

Code can be found here: https://scm.dazzlingwrench.fxinnovation.com/fxinnovation-public/terraform-module-aws-backup/pulls/2/files

### Expected Behavior

I expect all resources to be destroyed without issue.

### Actual Behavior

**Randomly**, Terraform fails to delete Backup Plan, saying Backup Selections should be deleted before:

```
Error: Error applying plan:

1 error occurred:
	* module.default.aws_backup_plan.this (destroy): 1 error occurred:
	* aws_backup_plan.this: error deleting Backup Plan: InvalidRequestException: Related backup plan selections must be deleted prior to backup plan deletion
	status code: 400, request id: 170f2b1c-156e-4e77-97e3-9aacbd9b6d02
```

Even if the actual order of deletion is correct:
```
module.default.aws_backup_selection.by_resources: Destruction complete after 1s
module.default.aws_backup_selection.by_tags: Destruction complete after 1s
aws_ebs_volume.example: Destroying... (ID: vol-0f6fecced036d1281)
module.default.aws_iam_role.this: Destroying... (ID: tftestbtsaul9fRole)
module.default.aws_backup_plan.this: Destroying... (ID: 170f2b1c-156e-4e77-97e3-9aacbd9b6d02)
```

My bet is that AWS API returns an OK status for Backup Selection deletion before it’s actually true…

### Steps to Reproduce

1. `terraform apply`
2. `terraform destroy`

**Comment by DrFaust92** [comment]
ill add a timeout for the `InvalidRequestException: Related backup plan selections must be deleted prior to backup plan deletion` error on delete to address this. but if there are selections that are not deleted for some reason (manually created or in a different module) it will just delay the error

**Comment by bflad** [comment]
A fix to shortly retry the deletion on this particular error has been merged and will release with version 3.20.0 of the Terraform AWS Provider. As mentioned above, this will only help for well-formed Terraform configurations. Thank you to @DrFaust92 for the implementation. 👍
